### PR TITLE
PROPOSAL: Bootstrap governance and simplified MyST EP process

### PR DESCRIPTION
**Pull request by choldgraf**
This is a proposal for a bootstrap governance model for the executablebooks organization, as well as a simplified MyST Enhancement Proposals model. Its goal is to give us the basic building blocks of explicit governance and decision making while being as simple as possible, in the hopes that we increase the likelihood that we can follow these processes.

It defines two different sources of truth.

- One for organizational policy, which would now live at a new repository called `team-compass`. We make changes to this repository via issues and PRs.
- One for the MyST specification, which lives at `myst-spec/`. We approve changes to this spec via a process carried out in `myst-eps`. This takes strong inspiration from [the MyST EPs proposal that @chrisjsewell put together](https://myst-eps.readthedocs.io/en/latest/mep-0001.html) and slims it down considerably in the hopes that it is more realistic for our (relatively) small team to follow.

In addition, it defines two decision-making groups:

- A **steering council**, that has decision-making power over organizational policy. Currently comprised of the three Principle Investigators of the original grant that funded this work.
- A **core team**, that has decision-making power over MEPs (and most decisions in most repositories) as delegated to by the Steering Council. This is roughly the @executablebooks/ebpteam (the folks defined [on the team page](https://executablebooks.org/en/latest/team.html))

My goal for this PR is just to get us on the same page, and to give us the basic building blocks we can use to start defining formal policy, strategy, etc.

My plan is to:

- Wait a bit for feedback, and incorporate ideas into the proposal
- Ask for approval from @gregcaporaso and @jstac 
- Once I have approval, I'll merge this in, and then:
- Create a `team-compass/` repository
- Move over a subset of information to encode there (e.g. team structure).
- We can then begin to iterate on more policy changes from there.
- Simultaneously, we will need to make the necessary changes to `myst-spec` and `myst-eps` to encode this process in those repositories as well.

### CODE REVIEW

**Comment by chrisjsewell** [comment]
Thanks @choldgraf this looks great, I'll have a proper look shortly 🙏

File: docs/governance-strategy.md
Comment (rowanc1) [inline docs/governance-strategy.md:131]:
```suggestion
   - A MEP may be accepted when all of the following conditions are met:
```

Just to be sure this isn't an `or`.

File: docs/governance-strategy.md
Comment (rowanc1) [inline docs/governance-strategy.md:132]: I think 2 days might be a bit short here, and I don't think we necessarily need that much speed in this aspect of our process. Perhaps 5-7 days as a cut off, which gives at least some time for reflection and discussion as opinions form and/or change.

File: docs/governance-strategy.md
Comment (gregcaporaso) [inline docs/governance-strategy.md:31]: Maybe a good place to add a reference to the code of conduct. For example:

```markdown
## Code of Conduct
Steering Council and Core Team Members agree to abide by the [EBP Code of Conduct](https://github.com/executablebooks/.github/blob/master/CODE_OF_CONDUCT.md). It is our expectation that _all contributors_ to the Executable Books Project (including but not limited to individuals contributing code, providing technical support, teaching workshops, and engaging in discussions about changes to EBP policy or the MyST Specification) adhere to the Code of Conduct.  
```

There's probably a better link to the CoC - I'm not sure what it is.

File: docs/governance-strategy.md
Comment (gregcaporaso) [inline docs/governance-strategy.md:44]: *aspects* -> *aspect*

File: docs/governance-strategy.md
Comment (rowanc1) [inline docs/governance-strategy.md:114]:
```suggestion
   ## UX Implications & Migration
```

I think calling out what migration is necessary should be in the header here, as it might be easy to overlook the description and also signals that we will support previous documents written in MyST over time.

File: docs/governance-strategy.md
Comment (gregcaporaso) [inline docs/governance-strategy.md:67]: *it may be merged any time.* -> *it may be merged any time, overriding the requirement of being open for five working days.*

(Just to be as clear as possible.)

File: docs/governance-strategy.md
Comment (rowanc1) [inline docs/governance-strategy.md:34]: I quite like this evolution, as it also gives us a chance to assess initial documentation that was created for a small team, and if those policies/processes are appropriate as the project evolves into a community effort.

File: docs/governance-strategy.md
Comment (gregcaporaso) [inline docs/governance-strategy.md:85]: Is this an SoT? It doesn't read like the others, which start by noting what they're a Source of Truth of. 

Based on my current read, it feels like there are two SoTs, and the _MyST Enhancement proposals_ section should be nested one more level (####) and its title aligned with _How to change team policy_ (e.g., _How to change the MysT Specification: MyST Enhancement proposals_)

File: docs/governance-strategy.md
Comment (rowanc1) [inline docs/governance-strategy.md:24]: `+1` from me on this. The steering council is consistent with the project reality at this stage (Project PIs), and this is also consistent from the bootstrap process laid out last September (https://github.com/executablebooks/meta/issues/493).

File: docs/governance-strategy.md
Comment (gregcaporaso) [reply docs/governance-strategy.md:85]: The second sentence on line 85 is missing a period at the end.

**Review by gregcaporaso — COMMENTED** [review]
Thanks so much for getting this started @choldgraf! I'll hold off on approving for right now, pending some discussion about whether or not it makes sense to mention the CoC in this document. My other comments are minor.

File: docs/governance-strategy.md
Comment (rowanc1) [inline docs/governance-strategy.md:93]: I think leaving off this number at the start? And it only gets a number when it is approved and merged? Perhaps this can be added below in the merge process.

File: docs/governance-strategy.md
Comment (rowanc1) [inline docs/governance-strategy.md:135]:
```suggestion
  - If the MEP is accepted, update its status metadata to `Accepted`, add an incremental ID (e.g. `id: 0002`) in the mep frontmatter, and merge the PR. Once a PR is merged, it closes the issue and a decision has been made.
```

File: docs/governance-strategy.md
Comment (rowanc1) [inline docs/governance-strategy.md:87]: I am a big fan of agreeing before we _formally_ agree on all the details, and the 2-4 co-authors also is important in ensuring that many avenues have been thought through from a few diverse angles.

**Review by rowanc1 — APPROVED** [review]
This looks great, and I think it is a good balance of pragmatism, the ability to evolve the process over time, and the rigour that we need now in some areas.

I do like that the MEP process is both light yet potentially quite rigorous, and is treated differently than the team process, which likely over the coming year may change substantially as the project evolves and has more of a community focus.

I am all for the current Steering Council and this reflects the reality of the grant funded project of today. I also trust that the PIs will evolve that processes effectively over time.

I have left a few minor comments and suggestions (`eps` name, increasing time in one part, a clarification, and a suggestion that we hold off on giving IDs until things are accepted). All of these are minor, and shouldn't hold up agreement without me (I am pretty busy this next week!).

Thanks for everyone's efforts on this, and looking forward to getting some version of this more formalized. :)

File: docs/governance-strategy.md
Comment (gregcaporaso) [reply docs/governance-strategy.md:132]: I thought that too when I first read it, but I haven't been involved with that process so didn't weigh in on it.

File: docs/governance-strategy.md
Comment (chrisjsewell) [reply docs/governance-strategy.md:24]: yep +1

File: docs/governance-strategy.md
Comment (mmcky) [reply docs/governance-strategy.md:83]: Agree that this is relatively minor point but I tend to agree with longer repo names (mainly thanks to the wonderful invention of tab completion) as it makes it clear what the repo contains.

File: docs/governance-strategy.md
Comment (mmcky) [reply docs/governance-strategy.md:132]: +1 I think a week is a good idea for public consideration of an MEP.

**Comment by rowanc1** [comment]
Another point to consider/document is the license of the MEPs, in @chrisjsewell's draft (I think based on PEPs?) I quite liked:

> This document is placed in the public domain or under the
> CC0-1.0-Universal license, whichever is more permissive.

I am not sure if that needs to be in each MEP, or is included in the license of the repository, but I think it is important to document/include in some respect. (CC-BY-4.0 could also be used potentially, and I am not sure if it would have too many differences/downsides in this context).

**Comment by chrisjsewell** [comment]
So the team-compass sounds great, I'll leave that to you guys

On the extension-proposal side, my main questions are:

1. As opposed to the current draft where there is three types of MEP: Standards Track, Informational, and Process, this only mentions the "Standards Track" one. Is it still the intention to have the three? Obviously, this actual document, of the MEP process would be a process one, or is intended that this sits outside the actual myst-eps (in the team-compass?).
  Their probably won't be so many Informational ones, but what I had envisaged (e.g. in https://github.com/executablebooks/myst-eps/pull/5) was having at least one with "core design principles/guidelines" that could/would be referenced in Standard MEPs, e.g. "does this syntax introduce multiple ways of doing the same thing", "does this syntax break commonmark".

2. The current terminology seems quite focussed on changes to current syntax. What about additions of syntax or (very likely) additions of new directives and roles? I could even think of extending this to proposing e.g. changes/additions to the options of directives, is this within the scope of a MEP?

3. > If the MEP is accepted, update its status metadata to `Accepted` and merge the PR. Once a PR is merged, it closes the issue and a decision has been made.

What does it mean that the decision is made, and how does it relate to actual implementations?

Most syntaxes are tied to reference implementations, e.g.:

- https://peps.python.org/pep-0001/#pep-review-resolution: Once a PEP has been accepted, the reference implementation must be completed. When the reference implementation is complete and incorporated into the main source code repository, the status will be changed to “Final”.

- https://jupyter.org/enhancement-proposals/jupyter-enhancement-proposal-guidelines/jupyter-enhancement-proposal-guidelines.html: If in the course of implementation, it is discovered that the implementation needs to be radically different from what was defined in the original JEP, then a pull request is submitted to modify the original JEP with the new necessary implementation, and a note citing the need for a modification to the JEP. This pull request should be re-approved by the original review team.

- https://github.com/commonmark/commonmark-spec: CommonMark is a rationalized version of Markdown syntax, with a [spec](http://spec.commonmark.org/) and BSD-licensed reference implementations in C and JavaScript.

Should there be a defined reference implementation for myst-spec, and should it be mystjs?
and/or there be an extra step in the MEP process, to allow for the implementation(s) and feedback from it?
What if a certain implementer, finds they cannot implement it fully or at all?
(I could certainly envisage this with myst-parser, and its need to adhere with sphinx/docutils)

Here this makes me think of the [web spec](https://www.w3.org/standards/) and https://caniuse.com, in that perhaps its ok that implementers don't implement all syntax (particularly all directives/roles), but then there is a way for them to provide a concise "manifest" of what they do support.

---

Perhaps this thought is a bit too technical for here:

Additionally, let's say we are targetting output formats HTML, PDF and Word.
Does the MEP have to discuss how the syntax can be rendered in each format?
Here I am thinking of thing like `tabs`, already implemented in both https://sphinx-design.readthedocs.io/en/furo-theme/tabs.html and https://js.myst.tools/guide/dropdowns-cards-and-tabs#tabs
would we envisage something like this becoming part of the myst-spec and, if so, add it in a MEP with guidance on what the behaviour should be for different

File: docs/governance-strategy.md
Comment (choldgraf) [reply docs/governance-strategy.md:31]: Agree that the Code of Conduct needs to be in a prominent place in any spot that describes the policies of the team. I think that we should have a link to the CoC in both the `executablebooks.org` and `compass.executablebooks.org` spot, but agreed that the "team compass" is where the Code of Conduct should be defined (or where it delegates the COC to another place)

File: docs/governance-strategy.md
Comment (choldgraf) [reply docs/governance-strategy.md:85]: Ha, that's how it was before and I changed it at the the last minute because I have a personal aversion to four-deep nested headers 😅, but it sounds like I should not have un-nested it after all. I'll change it back and I agree w/ you!

File: docs/governance-strategy.md
Comment (choldgraf) [reply docs/governance-strategy.md:132]: That sounds good to me - my first thought was to keep it short so that the Core Team felt they had the agility needed to evolve the spec rapidly if need be, but I'm definitely fine with lengthening that process. The MyST EPs process in particular is a broader team process, so I welcome suggestions from that team on iterative ways we can improve the text here.

File: docs/governance-strategy.md
Comment (choldgraf) [reply docs/governance-strategy.md:93]: How about once it becomes "Active" it gets a number. That way if a MEP is not accepted, it still has a number and goes on-record as not being accepted. This could help avoid re-hashing the same discussions multiple times because people didn't realize it had already been discussed and not accepted.

**Comment by chrisjsewell** [comment]
> a pull request to the myst-spec repository (or any implementations) will be non-controversial
> if the lack of implementation there creates too much tension or sluggishness

Yeh this is obviously the point I'm getting at, particularly from the view-point of myst-parser (or indeed stakeholders for any implementation):
If I see an MEP where my feeling is "I like the syntax, but oof this could be a nightmare to implement (particularly within the sphinx framework)", but I won't be clear until an actual implementation has been tried, do I:

1. Say no, this MEP cannot be merged until there is a PR for an implementation in myst-parser (which could possibly take a while)
2. Say yes, this MEP can be merged (and the spec changed), with the caveat that it can be modified or even revoked after an implementation is tried in myst-parser. During this intermediate time, then, myst-parser and thus jupyter-book would be non-spec compliant.
3.  Say yes, this MEP can be merged (and the spec changed), and if ultimately it cannot be implemented that is "ok" and myst-parser/jupyter-book are simply non-spec compliant in this area

I feel this situation could arise very quickly, and I'm very keen to have clarity here from the team from the outset

**Comment by choldgraf** [comment]
Why not use something similar to your proposal above:

> Here this makes me think of the [web spec](https://www.w3.org/standards/) and [caniuse.com](https://caniuse.com/), in that perhaps its ok that implementers don't implement all syntax (particularly all directives/roles), but then there is a way for them to provide a concise "manifest" of what they do support.

I'd hope the Core Team could raise concerns about implementation feasibility at the design level before the MEP proposal is accepted. But if we miss something that is only uncovered later at implementation, the person could:

1. Create a section of their documentation that lists specific ways in which their implementation _deviates_ from the MyST spec, but otherwise adheres to spec version X.x.
	- This would let implementors have a little bit of wiggle room, provided they weren't deficient in major functionality.
	- We should also make it clear in the spec repository (via a design / strategy doc like the one y'all are working on) that MyST should make choices that are not overfit to a specific implementation.
2. If it's really bad, create a new MEP to propose revoking it or modifying it to be more feasible. If there's a delay between when MEPs are accepted and when a new "version" of the MyST spec is released, there might also be some time to modify as-needed once implementation has happened. Ideally when a new version of the MyST spec is released, many implementations have _already_ tried making the necessary changes.

File: docs/governance-strategy.md
Comment (rowanc1) [reply docs/governance-strategy.md:93]: Yes, that sounds great. Good to have a record of things discussed and decided against.

File: docs/governance-strategy.md
Comment (fperez) [inline docs/governance-strategy.md:24]: "Principle" -> "Principal" (principle means something else)

**Comment by choldgraf** [comment]
I believe that I've addressed the comments above, and have also left a few comments and empty areas in-line with the markdown to show where we need to fill in information in future iterations.

I split the document into three pages to make it easier to parse, but the content hasn't meaningfully changed beyond the suggestions people gave.

I request approval from @jstac and @gregcaporaso, and once given I'll start creating a `team-compass` repository. I've encoded those next steps in an issue here:

- https://github.com/executablebooks/meta/issues/846

File: docs/bootstrap-governance/governance.md
Comment (gregcaporaso) [inline docs/bootstrap-governance/governance.md:4]: _adhere to_ -> _are expected to adhere to_ or _implicitly agree to abide by_

I think that's an important distinction as the former is a description of idealized behavior of the community, while the latter sets out our expectations for community members. (And apologies if I'm requesting a change to text I originally drafted! I'm reading on this subject now.)

**Review by gregcaporaso — APPROVED** [review]
I have one suggested modification, but approve whether that change is made or not as we can also get it in during the next stage of governance formalization.

**Review by jstac — APPROVED** [review]

**Comment by choldgraf** [comment]
OK I updated the text w/ the minor comments that remained. I see 2 approvals from the @executablebooks/steering-council (I just created that group by the way) so will merge in!

**Comment by gregcaporaso** [comment]
Thanks @choldgraf for leading this effort, and thanks everyone for the input!

**Comment by sarbo82** [comment]
CREATE TABLE "topic" (
    "id" integer NOT NULL PRIMARY KEY,
    "forum_id" integer NOT NULL,
    "subject" varchar(255) NOT NULL
);
ALTER TABLE "topic"
ADD CONSTRAINT forum_id FOREIGN KEY ("forum_id")
REFERENCES "forum" ("id");

-- Initials
insert into "topic" ("forum_id", "subject")
values (2, 'D''artagnian');

**Pull request merged**
